go.mod

```diff
@@ -68,6 +68,17 @@ require (
 	k8s.io/utils v0.0.0-20220210201930-3a6ce19ff2f9
 )
 
+require (
+	github.com/Masterminds/semver/v3 v3.1.1
+	github.com/cavaliergopher/grab/v3 v3.0.1
+	github.com/go-playground/validator/v10 v10.9.0
+	github.com/google/go-cmp v0.5.7
+	github.com/robfig/cron v1.2.0
+	gopkg.in/yaml.v2 v2.4.0
+	k8s.io/apiextensions-apiserver v0.24.0
+	k8s.io/kubernetes v1.13.0
```

Comment:
This doesn't look right (`go mod tidy` is trying to trick you)

I'd expect this to be `v1.24.0`, and doing that there should be a number of `replace` entries for each of the submodules pinning them to `v1.24.0`. This was all needed just to use the k8s ready-probing code.

From past experiences with this, I expect it to be tricky.

Reply:
> It's the version we use in the k0s currently.

You mean in autopilot, not in k0s? In k0s, we definitely use 1.24.0.

Since the code is currently not yet active and k8s.io/kubernetes is just used in a single file for a HTTP probe, what about commenting that piece of code out and remove the dependency, until we can re-add it properly?

Reply:
Life will be much easier if you reference the `require` + `replace` directives in autopilot as this work has already been done.

Autopilot uses `k8s.io/kubernetes v1.24.0` -- `go mod tidy` is picking up the wrong version dependency because of the lack of replacement entries.

Reply:
@twz123 no, we don't: https://github.com/k0sproject/k0s/blob/main/go.sum#L2291

I'm a bit losing the point of having a separate branch and keeping PRs small if we start making random improvements. In this PR I've put the code of autopilot packages, it doesn't pass our checks and doesn't work, so I'd suggest doing all the improvements in the separate small PRs. Otherwise, it will impossible to track the changes.

Reply:
This still needs to be `k8s.io/kubernetes v1.24.0` in `go.mod`.

The failing unit tests are the reason why
https://github.com/k0sproject/k0s/runs/6569146415?check_suite_focus=true#step:5:127

Reply:
So there are 4 PRs in motion (#1732, #1733, #1737, and #1739) with ~150 changed files that are nearly identical with each other, where three PRs are used to fix shortcomings with the first uncommitted PR? I don't understand this approach, or even how others can work with it.

I can't do anything until this stack of PRs is unwound.